conda_smithy/configure_feedstock.py

```diff
@@ -224,9 +276,9 @@ def break_up_top_level_values(top_level_keys, squished_variants):
                     #    config in its own file
 
                     zipped_config = []
-                    top_level_config_dict = OrderedDict()
+                    top_level_config_dict: dict = OrderedDict()
                     for idx, variant_key in enumerate(squished_variants[key]):
-                        top_level_config = []
+                        top_level_config: Any = []
```

Comment:
I am sure you checked this, but can we avoid `Any`?

Reply:
this is a tricky one:( `top_level_config` is being declared as a `list` initially and then being converted to `tuple` on line 287. i tried using `list | tuple` but then it throws another error for `append`.
`Item "tuple[Any, ...]" of "list[Any] | tuple[Any, ...]" has no attribute "append"`
I added `Any` to avoid the errors since i couldn't find any other appropriate type 🙈

Reply:
If something is being converted to a different type, you could try using a new variable name when the conversion happens and giving each variable a distinct type.